x-pack/plugins/task_manager/server/task_store.ts

```diff
@@ -345,7 +345,7 @@ export class TaskStore {
    */
   public async remove(id: string): Promise<void> {
     try {
-      await this.savedObjectsRepository.delete('task', id);
+      await this.savedObjectsRepository.delete('task', id, { refresh: false });
```

Comment:
Without `refresh: false`, it would cause ad-hoc tasks to take up to 1s longer (pr however long the next refresh is) before task manager frees up a worker for the next task.